doc/development_process/index.rst

```diff
@@ -0,0 +1,301 @@
+    respective maintainers.)
+  - When the merge window is closed, the gatekeeper lays a vN-rc1 tag and the tree
+    enters the release candidate phase
+  - CI sees the tag, builds and runs tests; QA analyses the report from the
+    build and test run and gives an ACK/NAK to the build
+  - The gatekeeper, with QA and any other needed input, determines if the release
+    candidate is a go for release
+  - If it is a go for a release, the gatekeeper lays a tag release vN at the same
+    point
+- Development on new features continues in feature branches. Once features are
+  ready, they are submitted to mainline during the merge window period and after
+  the release is tagged.
+
+.. figure:: release_cycle.png
+    :align: center
+    :alt: Release Cycle
+    :figclass: align-center
+
+    Release Cycle
+
+Merge Window
+*************
+
+A relatively straightforward discipline is followed with regard to the merging
+of patches for each release.  At the beginning of each development cycle, the
+"merge window" is said to be open.  At that time, code which is deemed to be
+sufficiently stable (and which is accepted by the development community) is
+merged into the mainline tree.  The bulk of changes for a new development cycle
+(and all of the major changes) will be merged during this time.
+
+The merge window lasts for approximately two months.  At the end of this time,
+the gatekeeper will declare that the window is closed and release the first of
+the release candidates.  For the codebase release which is destined to be 0.4.0,
+for example, the release which happens at the end of the merge window will be
+called 0.4.0-rc1.  The -rc1 release is the signal that the time to merge new
+features has passed, and that the time to stabilize the next release of the code
+base has begun.
+
+Over the next weeks, only patches which fix problems should be submitted to the
+mainline.  On occasion, a more significant change will be allowed, but such
+occasions are rare and require a TSC approval (Change Control Board). As a
+general rule, if you miss the merge window for a given feature, the best thing
+to do is to wait for the next development cycle.  (An occasional exception is
+made for drivers for previously unsupported hardware; if they do not touch any
+other in-tree code, they cannot cause regressions and should be safe to add at
+any time).
+
+As fixes make their way into the mainline, the patch rate will slow over time.
+The mainline gatekeeper releases new -rc drops once or twice a week; a normal
+series will get up to somewhere between -rc4 and -rc6 before the code base is
+considered to be sufficiently stable and the final 0.4.x release is made.
+
+At that point, the whole process starts over again.
+
+.. figure:: merge_window.png
+    :align: center
+    :alt: Merge Window
+    :figclass: align-center
+
+    Merge Window
+
+Here is the description of the various moderation levels:
+
+- Low:
+
+  - Major New Features
+  - Bug Fixes
+- Refactoring
```

Comment:
Shouldnt refactoring be indented in the same level as bugfixes and structure directory changes?